torch/csrc/api/include/torch/nn/options/activation.h

```diff
@@ -56,5 +56,15 @@ struct LeakyReLUOptions {
   TORCH_ARG(bool, inplace) = false;
 };
 
+// ============================================================================
+
+/// Options for the `Softmax` module.
+struct TORCH_API SoftmaxOptions {
+  // Dimension along which Softmax will be computed.
+  TORCH_ARG(int, dim);
```

Comment:
Please make it constructor argument with default value, as @yf225 asked [here](https://github.com/pytorch/pytorch/pull/27035#discussion_r329657107), see HardshrinkOptions as an example

Reply:
I suggest to have -1 as a default value

Reply:
Thanks. I fixed it.

Reply:
After discussion with the team we agreed that we should use c10::optional<int64_t> instead of just int here. The default value should be c10::nullopt. I'm sorry for for misleading you.

Reply:
@pbelevich 

No problem. Thanks for the instruction.
Does this rule also apply to `torch::Dtype` ?

Reply:
`dim` should be a non-optional argument because allowing it to be null is already deprecated in Python version, and the C++ version should just match the newest Python version behavior.

Reply:
Thanks, I fixed it to make `dim` non-optional argument.